tests/test_footprint1.py

```diff
@@ -682,6 +682,16 @@ def testSetFromFootprint(self):
 
         self.assertTrue(objects[1].contains(afwGeom.Point2I(3, 6)))
 
+        # Verify the FootprintSet footprint list setter can accept inputs from
+        # the footprint list getter
+        getterSetterTestOK = True
+        try:
+            ds.setFootprints(ds.getFootprints())
+        except:
+            getterSetterTestOK = False
+        self.assertTrue(getterSetterTestOK, "The FootprintSet footprint "
+                        "setter can't accept getter output as an input")
```

Comment:
As we discussed, I find the above a little convoluted: I think it would be more straightforward to simply let the exception propagate rather than catch it and `assertTrue`. While I realise you're trying to print a helpful message, I don't think expecting somebody debugging a test failure to look at a comment in the source to understand what went wrong is unreasonable.

I also think we could be a bit more ambitious with the test. Demonstrating that the setter doesn't throw when called with the output of the getter is necessary, but it's not sufficient behaviour for the code: we really want to see that `setFootprints` really does set the footprints when it's called. As far as I can see, this function isn't tested anywhere else (if it was, we'd probably never have hit this error), so dropping in a real test for it now seems worthwhile.

Off the top of my head, I would:

- Retrieve a list of footprints from `ds.getFootprints()`;
- Perform some transformation on that list (e.g. just delete the last one, assuming there are multiple footprints in the list);
- Feed that to `setFootprints()`;
- Check that calling `getFootprints()` gets you back the modified list.

That probably takes more characters to write in English than it does in Python, so I shouldn't think it'd be a big effort. Am I wrong?